lib/connectors_async/job_runner.rb

```diff
@@ -63,16 +64,15 @@ def log_with_thread_id(level, str)
     private
 
     def with_throttling(job)
-      attempts = 0
       if job.should_wait?
         log_with_thread_id(:info, "Job #{job.id} is sleeping: Enterprise Search hasn't picked up documents for a while.")
-
+        wait_duration = 0
         while job.should_wait?
-          if attempts < ConnectorsShared::Constants::MAX_IDLE_ATTEMPTS
-            attempts += 1
+          if wait_duration < ConnectorsShared::Constants::MAX_IDLE_TIME
             idle(ConnectorsShared::Constants::IDLE_SLEEP_TIME)
+            wait_duration += ConnectorsShared::Constants::IDLE_SLEEP_TIME
```

Comment:
Looks a bit awkward, but I believe it's fine - we can actually calculate the idling duration, which will add some milliseconds, but on a high level it should work, unless `sleep` actually sleeps for much more than expected - can happen under heavy load.

Should we address it, team?

Reply:
I think you can just record the idle start time. Then compare `Time.now - idle_start_time` with `ConnectorsShared::Constants::MAX_IDLE_TIME`. Assuming `ConnectorsShared::Constants::IDLE_SLEEP_TIME` in time passed is inaccurate, as you note.

Reply:
`Time.now - idle_start_time` makes more sense.